azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java

```diff
@@ -340,8 +340,13 @@ public Job buildJobExecutor(String jobId, Props jobProps, Logger logger)
             jobProps, jobType));
       }
 
-      // TODO: should the logic below mirror the logic for PluginLoadProps?
       Props pluginJobProps = pluginSet.getPluginJobProps(jobType);
+      // For default jobtypes, even though they don't have pluginJobProps configured,
+      // they still need to load properties from common.properties file if it's present
+      // because common.properties file is global to all jobtypes.
+      if(pluginJobProps == null) {
+        pluginJobProps = pluginSet.getCommonPluginJobProps();
+      }
       if (pluginJobProps != null) {
```

Comment:
The if clause starting in line 350 actually applies to both default and custom plugin jobtypes. If changed to if else structure, then for default jobtypes, this part will be skipped and jobProps will not be populated for default jobtypes.